x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/jinaai/rerank/JinaAIRerankServiceSettings.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.services.jinaai.rerank;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.common.ValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.inference.ModelConfigurations;
+import org.elasticsearch.inference.ServiceSettings;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.inference.services.ConfigurationParseContext;
+import org.elasticsearch.xpack.inference.services.jinaai.JinaAIRateLimitServiceSettings;
+import org.elasticsearch.xpack.inference.services.jinaai.JinaAIServiceSettings;
+import org.elasticsearch.xpack.inference.services.settings.FilteredXContentObject;
+import org.elasticsearch.xpack.inference.services.settings.RateLimitSettings;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.xpack.inference.services.ServiceFields.DIMENSIONS;
+import static org.elasticsearch.xpack.inference.services.ServiceFields.MAX_INPUT_TOKENS;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractSimilarity;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.removeAsType;
+
+public class JinaAIRerankServiceSettings extends FilteredXContentObject implements ServiceSettings, JinaAIRateLimitServiceSettings {
+    public static final String NAME = "jinaai_rerank_service_settings";
+
+    private static final Logger logger = LogManager.getLogger(JinaAIRerankServiceSettings.class);
+
+    public static JinaAIRerankServiceSettings fromMap(Map<String, Object> map, ConfigurationParseContext context) {
+        ValidationException validationException = new ValidationException();
+        var commonServiceSettings = JinaAIServiceSettings.fromMap(map, context);
+        // We need to extract/remove those fields to avoid unknown service settings errors
+        extractSimilarity(map, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        removeAsType(map, DIMENSIONS, Integer.class);
+        removeAsType(map, MAX_INPUT_TOKENS, Integer.class);
```

Comment:
I think this is not needed. I guess you've looked at `CohereRerankServiceSettings`? These rerank service settings contained text embedding service settings in the past (which was wrong), therefore we had to do this dance in `CohereRerankServiceSettings` to maintain backwards-compability while fixing the bug.

Reply:
Yes I noticed that and did proper refactoring of my Service Settings inheritance trees. However I just mantained it in case it was necessary. I can remove these

Reply:
Thanks 🙏